sandy/fy.py

```diff
@@ -445,24 +438,21 @@ def gls_update(self, zam, e, decay_data, xp, Vy=None, kind='cumulative',
 
         """
         data = self.data
-        S = _gls_setup(decay_data, kind, index_columns)
         # Filter FY data:
         fy_data = self.filter_by('ZAM', zam)\
                       .filter_by("E", e).data\
                       .set_index('ZAP')[['MT', 'FY', 'DFY']]
         # Divide the data:
         fy_type = fy_data.groupby('MT')
         IFY, CFY = [fy_type.get_group(x)['FY'] for x in fy_type.groups]
-        DIFY, DCFY = [fy_type.get_group(x)['DFY'] for x in fy_type.groups]
+        mask = (data.ZAM == zam) & (data.MT == 454) & (data.E == e)
+        DIFY = data.loc[mask, 'DFY']
+        # Fix the S with the correct dimension:
+        S = _gls_setup(decay_data, kind, index=IFY.index, columns=IFY.index)
```

Comment:
Here we are again! the sensitivity is MxN, not MxM. Then index and columns in general are not the same! It looks this concept is not clear!

in the linear model `y=S*x`, where `y is Mx1` and `x is Nx1`, `S` must be `MxN`! nobody says that y and x have the same dimensions (and the same index)!

Another examples, i consider a fuel decay heat model that depends on the independent yields, something like `DH=f(IFY)` where `DY` is a scalar value in Watt, i.e., the fuel total decay heat. Do y and x have the same index? NO!